### PR TITLE
Mostly rewrite `esp-hal/README.md` (oops :) )

### PR DESCRIPTION
**Pull request by jessebraham**
So uh, sorry 😅  Didn't mean to make this many changes but, well... here we are 😁 

[Rendered view](https://github.com/JurajSadel/esp-hal/blob/b87e5a25d8af1c1520bcac9983724994377a5730/esp-hal/README.md)

### CODE REVIEW

File: esp-hal/README.md [hunk 1]
```diff
@@ -5,49 +5,51 @@
...
+[documentation]: https://docs.rs/esp-hal/
 
 ## Usage
 
-Before using this `esp-hal`, be sure you [configured] the environment and [tooling] correctly.
-We **highly recommend** to use our [esp-template] that does all the basic configuration for you. If you want to try to run examples, the [xtask] should be used:
+Before using `esp-hal`, ensure that you have configured your [development environment] correctly, and the [required tooling] has been installed.
 
-`cargo xtask run-example examples esp32c3 hello_world`
+When starting a new project using `esp-hal`, we strongly recommend you generate a project skeleton using [cargo-generate] and [esp-template]. This will take much of the guesswork out of the process and give you a starting point to build an application form.
```
Comment (JurajSadel) [inline esp-hal/README.md:24]: Should it be form or from? :D

**Review by JurajSadel — COMMENTED** [review]
LGTM, just one question.

File: esp-hal/README.md (hunk 1 above)
Comment (jessebraham) [reply esp-hal/README.md:24]: From :D

**Review by JurajSadel — APPROVED** [review]
Thanks, much better than my version :D

File: esp-hal/README.md (hunk 1 above)
Comment (JurajSadel) [reply esp-hal/README.md]: Haha, but even form is not that bad here, or is it? :D

**Pull request merged**